### DISCUSSION TITLE
device as an input parameter for the device node

### DISCUSSION THREAD

**Discussion by nalberto** [thomassorensen2650/node-red-contrib-mqtt-sparkplug-plus#33] [Ideas]
I would really need the device as an input  on the sparkplug device node if possible.
My flow is designed to work with multiple devices (corresponds to sparkplug edge of node)

**Comment by thomassorensen2650** [comment]
Hi.

Is the request to be able to set the broker connection for a device via a input command message (the same way do for metric definitions currently), Is  or is the request to be able to dynamically create (and delete?) devices via a input message?

**Comment by nalberto** [comment]
Hi,

Yes to dynamically create devices via an input message.

Thank you

**Comment by thomassorensen2650** [comment]
That seems like a pretty useful feature - I think I have a an idea on how to implement this with very little code changes.. All we need is a new rename device command (see example below):

```javascript
msg = {
      "command" : {
        "device" : {
            "rename" : "My Device Name"
        }
    } ,  
    "definition" : {
        "TEST/TEST" : {
            "dataType" : "Int32",
            "properties": {
                "engUnits": {
                    "type": "String",
                    "value": "inHg"
                }
			}, 
        }
    }
```

In the newest version (not released yet) a device will not birth when you haven't configure any metrics. If you create x amount (x = the max amount of devices you would ever need) of devices without any metrics, then you can send a definition and a rename-device command message to name the device and define metrics. 

This approach will also make it easy to troubleshoot via the node-red editor. Would this approach work for your usecase?

**Comment by nalberto** [comment]
I was planning to use it to send data to multiple end of nodes devices dynamically.
Would that still work do you think and keep record of the definitions for each device created. Maybe I am pushing it a bit but that was my use case. I would use it to send the definition and data for each new device discovered.

If that is not possible the rename would still be very usefull for me. I could maybe keep track of the definition separately and rename the device before sending the definition for device_x.

**Comment by thomassorensen2650** [comment]
Device rename has been added in PR #37 will be added to next release. I will also see if I can get EON (node) rename implemented before next release.

**Comment by nalberto** [comment]
Great. Thank you for your help on this.
